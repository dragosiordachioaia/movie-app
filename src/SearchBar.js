import React, { Component } from "react";

import "./SearchBar.scss";

export default class SearchBar extends Component {
  constructor(props) {
    super(props);

    // this.displayStuff = this.displayStuff.bind(this);
  }

  displayStuff() {}

  render() {
    return <div> SearchBar </div>;
  }
}
