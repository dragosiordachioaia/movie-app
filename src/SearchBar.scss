@import "variables";

.search-bar-container {
  width: calc(100% - #{$sidebar-width});

  .search-bar {
    margin-top: 0.4em;
    margin-left: 2em;
  }

  i {
    margin-right: 10px;
  }

  input {
    background-color: transparent;
    border: 0;
    color: rgba(255, 255, 255, 1);
    width: calc(100% - 8em);
    padding: 0.4em;
    font-size: 0.8em;
    font-weight: 700;
    outline: 0;
    transition: border 300ms;
    border-bottom: 1px solid transparent;

    &:focus {
      border-bottom: 1px solid rgba(255, 255, 255, 0.7);
    }

    &::placeholder {
      color: rgba(255, 255, 255, 0.6);
      font-weight: 700;
    }
  }
}
