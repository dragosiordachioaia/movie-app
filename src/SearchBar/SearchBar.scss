@import "../variables";

.search-bar-container {
  width: calc(100% - #{$sidebar-width});
  position: relative;

  .search-bar {
    left: 0.5em;
    transform: translateY(-50%);
    top: 50%;
    position: absolute;
    width: calc(100% - 3em);

    @media (min-width: $SM) {
      left: 1em;
    }
  }

  .fa {
    position: absolute;
    transform: translateY(-50%);
    top: 50%;
  }

  input {
    background-color: transparent;
    border: 0;
    color: white;
    width: 100%;
    font-size: 1em;
    font-weight: 700;
    outline: 0;
    transition: border 300ms;
    border-bottom: 1px solid transparent;
    position: absolute;
    left: 1.2em;
    transform: translateY(-50%);
    top: 50%;
    border-radius: 0;

    &:focus {
      border: 0;
      // border-bottom: 1px solid rgba(255, 255, 255, 0.7);
    }

    &::placeholder {
      color: rgba(255, 255, 255, 0.6);
      font-weight: 700;
    }

    @media (min-width: $SM) {
      left: 1.6em;
    }
  }
}
