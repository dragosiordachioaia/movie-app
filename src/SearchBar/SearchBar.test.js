import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import SearchBar from "./SearchBar";

let component;
let instance;

beforeEach(() => {
  component = renderer.create(<SearchBar />);
  instance = component.getInstance();
});

it("renders correctly", () => {
  instance.props = {};
  const result = instance.render();
  expect(result).toMatchSnapshot();
});
