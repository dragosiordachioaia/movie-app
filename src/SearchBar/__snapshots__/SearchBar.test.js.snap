// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="search-bar"
>
  <i
    className="fa fa-search"
  />
  <input
    onChange={[Function]}
    placeholder="Filter by name"
    type="text"
  />
</div>
`;
